### PR TITLE
Cert create user list

### PR DESCRIPTION
**Pull request by d1str0**
it's so fucking late... now you can create certs in the ui. give it a shot

### CODE REVIEW

**Comment by mattklepp** [comment]
This change is fine. I'd approve it if I were a reviewer.

File: ui/ui.go [hunk 1]
```diff
@@ -62,10 +62,12 @@ func (h *uiHandler) Route(unsafe bool) http.Handler {
 
 	r := mux.NewRouter()
 	r.HandleFunc("/ui/dashboard", h.Authenticated(h.Dashboard()))
-	r.HandleFunc("/ui/cert", h.Authenticated(h.ListCertificates()))
-	r.HandleFunc("/ui/cert/{id}", h.Authenticated(h.ViewCertificate())).Methods("GET")
-	r.HandleFunc("/ui/cert/{id}/edit", h.Authenticated(h.EditCertificate())).Methods("GET")
-	r.HandleFunc("/ui/cert/{id}/edit", h.Authenticated(h.SaveCertificate())).Methods("POST")
+	r.HandleFunc("/ui/certificates", h.Authenticated(h.ListCertificates()))
```
Comment (debus) [inline ui/ui.go]: Should this not only be allowed for GET?

File: ui/ui.go [hunk 2]
```diff
@@ -62,10 +62,12 @@ func (h *uiHandler) Route(unsafe bool) http.Handler {
...
+	r.HandleFunc("/ui/certificate/id/{id}", h.Authenticated(h.ViewCertificate())).Methods("GET")
+	r.HandleFunc("/ui/certificate/id/{id}/edit", h.Authenticated(h.EditCertificate())).Methods("GET")
+	r.HandleFunc("/ui/certificate/id/{id}/edit", h.Authenticated(h.SaveCertificate())).Methods("POST")
+
+	r.HandleFunc("/ui/certificate/create", h.Authenticated(h.CreateCertificate()))
```
Comment (debus) [inline ui/ui.go]: This should probably only be for POST requests

File: ui/ui.go [hunk 3]
```diff
@@ -234,6 +236,111 @@ func (h *uiHandler) Dashboard() http.HandlerFunc {
...
+
+type certValidation struct {
+	Domains string
+	RenewAt string
+	Email   string
+	Success string
+	Error   string
+}
+
+// Serve /ui/certificate/{id} page.
```
Comment (debus) [inline ui/ui.go]: Nitpicky but this actually serves `/ui/certificate/create`

File: ui/ui.go (hunk 2 above)
Comment (debus) [reply ui/ui.go]: Oh I see this URL is also called via POST for form submission, with the creation UI being served via GET. Can we mark this as serving both GET and POST?

**Review by debus — DISMISSED** [review]
One comment nitpick and one suggestion about more explicitly defining what methods a URL endpoint supports but other than that it looks great!

**Pull request merged**